endpoint.go

```diff
@@ -361,7 +365,18 @@ func (ep *endpoint) Leave(sbox Sandbox, options ...EndpointOption) error {
 		return err
 	}
 
-	return sb.clearNetworkResources(ep)
+	if err := sb.clearNetworkResources(ep); err != nil {
+		return err
+	}
+
+	if sb.needDefaultGW() {
+		ep := sb.getEPwihtoutGateway()
```

Comment:
I am not sure, but if an endpoint leave call can happen in another thread in between line 372 and 373, then we should not report an error.

You may want to check this code logic under concurrent calls.